py/pysparkling/initializer.py

```diff
@@ -65,15 +65,21 @@ def __get_sw_jar():
             from pkg_resources import get_cache_path
             zip_file = sw_pkg_file[:-len('/sparkling_water/__init__.py')]
             zip_filename = os.path.basename(zip_file)
-            jar_path = get_cache_path(zip_filename)
-            sw_jar = os.path.abspath("{}/sparkling_water/sparkling_water_assembly.jar".format(jar_path))
-            import zipfile
-            with zipfile.ZipFile(zip_file) as fzip:
-                fzip.extract('sparkling_water/sparkling_water_assembly.jar', path = jar_path)
-            return sw_jar
+            cache_path = get_cache_path(zip_filename)
+            cached_jar = os.path.abspath("{}/sparkling_water/sparkling_water_assembly.jar".format(cache_path))
+            if os.path.exists(cached_jar) and os.path.isfile(cached_jar):
```

Comment:
Thinking about it more, can we disable caching totally? It could be dangerous, if we have cached old version of jar and upgraded sw.